crates/indexer/src/indexer.rs

```diff
@@ -87,364 +104,415 @@ impl Web3Indexer {
+
+                    let nonce: u32 = l2_transaction.raw().nonce().unpack();
+
+                    let tx_receipt: TxReceipt =
+                        self.get_transaction_receipt(gw_tx_hash, block_number, tx_index)?;
+
+                    let exit_code: u8 = tx_receipt.exit_code().into();
+                    let web3_transaction = Web3Transaction::new(
+                        gw_tx_hash,
+                        None,
+                        block_number,
+                        block_hash,
+                        tx_index,
+                        from_address,
+                        Some(to_address),
+                        value,
+                        nonce,
+                        gas_limit,
+                        gas_price,
+                        Vec::new(),
+                        r,
+                        s,
+                        v,
+                        0, // should update later
+                        gas_limit,
+                        None,
+                        exit_code,
+                    );
+
+                    let web3_tx_with_logs = Web3TransactionWithLogs {
+                        tx: web3_transaction,
+                        logs: vec![],
+                    };
+
+                    return Ok(Some(web3_tx_with_logs));
                 }
-                tx_index += 1;
+                SUDTArgsUnion::SUDTQuery(_sudt_query) => {}
             }
+            // tx_index += 1;
         }
-        Ok(web3_tx_with_logs_vec)
+        Ok(None)
+    }
+
+    async fn batch_from_script(
+        &self,
+        txs: &[L2Transaction],
+    ) -> Result<std::collections::HashMap<u32, Option<Script>>> {
+        let from_ids = txs
+            .iter()
+            .map(|tx| {
+                let from_id: u32 = tx.raw().from_id().unpack();
+                from_id
+            })
+            .collect::<Vec<_>>();
+        let from_ids_len = from_ids.len();
+
+        let mut to_ids = txs
+            .iter()
+            .map(|tx| {
+                let to_id: u32 = tx.raw().to_id().unpack();
+                to_id
+            })
+            .collect::<Vec<_>>();
+        let to_ids_len = to_ids.len();
+
+        let mut ids = from_ids;
+        ids.append(&mut to_ids);
+
+        let id_sets: HashSet<u32> = std::collections::HashSet::from_iter(ids.into_iter());
+        let ids = Vec::from_iter(id_sets.into_iter());
+        let ids_len = ids.len();
+
+        log::debug!(
+            "batch_from_script request, from_id len: {}, to_id len: {}, id sets len: {}",
+            from_ids_len,
+            to_ids_len,
+            ids_len
+        );
+
+        let scripts = batch_account_id_to_script(&self.godwoken_async_client, ids.clone()).await?;
+
+        let mut hashmap = HashMap::<u32, Option<Script>>::new();
+        scripts.into_iter().enumerate().for_each(|(i, value)| {
+            let id = ids[i];
+            hashmap.insert(id, value);
+        });
```

Comment:
[`zip`](https://doc.rust-lang.org/1.61.0/core/iter/fn.zip.html) may be more fitable here.

Reply:
update